### DISCUSSION TITLE
XModem Bootloader for the DE0-Nano

### DISCUSSION THREAD

**Discussion by emb4fun** [stnolting/neorv32#594] [Show and tell]
Hello,

with this project it should be possible to load a program into one of the different memories and start it there:

- XIP (Execute in Place memory, existing SPI flash on the DE0-Nano)
- IMEM (TCM, Tightly Coupled Memory)
- SDRAM
- SD Card to SDRAM (under development)

In addition, it should also be possible to update the FPGA image.

A detailed description of the project can be found on my website here:
https://www.emb4fun.de/archive/neorv32xboot/index.html

Best regards,
Michael

**Comment by stnolting** [comment]
This is really awesome! The XBOOT format seems to be widely adopted and with your bootloader we are one step closer making integration of this core (even?) more generic and user-friendly.

Thank you very much Michael for your great work! :+1:
